### PR TITLE
Search: add API data for record meter

### PR DESCRIPTION
**Pull request by annacmc**
This PR connects the record meter chart up with real API data

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* On a site with an active Jetpack Search subscription, go to  `/wp-admin/admin.php?page=jetpack-search&features=record-meter`
* Check that the record meter chart data is accurate to the website (eg. number of posts, pages, plan tier limit) are correct, and if you add/remove posts/pages/custom post types that this is reflected in the chart. 

image

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js [hunk 1]
```diff
@@ -6,17 +6,14 @@ const PALETTE = require( '@automattic/color-studio' );
...
- * @param {object} planInfo - information about the users current plan
+ * @param {object} data - contains record type breakdown, post count, last indexed date, and tierMaximumRecords
  * @returns {object} data in correct form to use in chart and notice-box
  */
-export default function getRecordInfo( data, planInfo ) {
-	// set max number of record types to display
-	const maxRecordCount = 5;
-
-	// sets up some basic counts and arrays
+export default function getRecordInfo( data ) {
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js:12]: `data` is very generic and doesn't give us much information at a glance when we look at the function signature.

As these inputs should stay pretty consistent, I would be tempted to provide these as separate arguments to the function (or at least destructure them like we would with component props).

File: projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js [hunk 2]
```diff
@@ -6,17 +6,14 @@ const PALETTE = require( '@automattic/color-studio' );
...
+	const maxRecordCount = 5; // this value determines when to cut off displaying post times & compound into an 'other'
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js:13]: Possibly `maxPostTypeCount` instead?

**Review by kangzj — DISMISSED** [review]
It works well for me and generally looks good to me 😄 Good job @annacmc 👍 

Left few comments but none of them are blockers or musts - just some advice for future consideration.

**Comment by annacmc** [comment]
> Left few comments but none of them are blockers or musts - just some advice for future consideration.

Thank you! This is really helpful feedback. I will definitely address this & think about it all in a future PR :)

**Review by kangzj — APPROVED** [review]

**Pull request merged**